### ISSUE TITLE
Standardize the use of calling super

### ISSUE THREAD

**Issue by bjschoenfeld** [byu-dml/d3m-dynamic-neural-architecture#47]

**Comment by orionw** [comment]
I assume this is because there are two different ways?  Which way do you want?

`RegressionModelBase.__init__(self, seed=seed)`
or 
`super(RNNDataset, self).__init__(data, features_key, target_key, y_dtype, device)`?

Those are the only two types I can find.  It looks like they're equivalent.
See https://stackoverflow.com/questions/222877/what-does-super-do-in-python/33469090#33469090 (second answer is best) and https://stackoverflow.com/questions/576169/understanding-python-super-with-init-methods (2nd answer)

Basically using super does the call implicity by figuring out the class to call whereas using it by saying `RegressionModelBase` makes it explicit.

**Comment by bjschoenfeld** [comment]
Thanks for the links. There is a third option

`super().__init__(*args, **kwargs)`

where the class is not specified. This does not work in Python 2, which is fine with me. I think I like this one best because it is easiest to maintain. All the base classes will be called, so if we change which base classes the class has or just rename them, we don't have to also change them here. If we change the name of the current class, we don't also have to change its name inside `super`.

We should use the `BaseClass.__init__` syntax when we know we don't want all base classes, but just a particular one.

**Comment by orionw** [comment]
Hey Erik, I didn't see you had self assigned this sorry.  I just made a PR for it.

**Comment by erikhuck** [comment]
Thanks for doing my work for me! Why did you reopen the issue though?

**Comment by orionw** [comment]
Yeah sorry about that haha.  I accidentally clicked "close and comment" instead of just comment.  The PR isn't merged yet.
